tests/test_locking.py

```diff
@@ -0,0 +1,79 @@
+"""
+Tests for our custom array writing locking stuff.
+"""
+
+import logging
+import sys
+from time import sleep
+
+import dask
+import numpy as np
+import pytest
+import zarr
+from dask.distributed import Client
+
+from pangeo_forge.utils import chunk_bounds_and_conflicts, lock_for_conflicts
+
+
+@pytest.mark.parametrize("n_tasks, conflicts", [(2, (0,)), (2, (1, 2)), (4, (0,))])
+def test_locks(n_tasks, conflicts, tmp_target, dask_cluster):
+
+    this_client = Client(dask_cluster)
+
+    @dask.delayed
+    def do_stuff(n):
+        logger = logging.getLogger("pangeo_forge")
+        handler = logging.StreamHandler(stream=sys.stdout)
+        handler.setLevel(logging.DEBUG)
+        logger.addHandler(handler)
+        with lock_for_conflicts(conflicts):
+            # test for concurrency by writing an object and asserting there are
+            # no other objects present
+            mapper = tmp_target.get_mapper()
+            key = f"foo_{n}"
+            mapper[key] = b"bar"
+            assert list(mapper) == [key]
+            sleep(0.25)
+            del mapper[key]
+            assert list(mapper) == []
+
+    dask.compute([do_stuff(n) for n in range(n_tasks)])
```

Comment:
This PR revealed that the locking I thought I had implemented was actually broken. Here is a test that really confirms it works.